### ISSUE TITLE
Does not install with ghc-7.10

### ISSUE THREAD

**Issue by NobbZ** [jaspervdj/stylish-haskell#77]
When trying to install/compile with ghc 7.10.0.20141227 I get the following errormessage.

```
src/Language/Haskell/Stylish/Config.hs:123:17:
    No instance for (FromJSON (t0 A.Value))
      arising from a use of ‘A..:’
    In the first argument of ‘(>>=)’, namely ‘o A..: "steps"’
    In a stmt of a 'do' block:
      steps <- (o A..: "steps"
                >>= fmap concat . mapM (parseSteps config))
    In the expression:
      do { config <- Config <$> pure [] <*> (o A..:? "columns" A..!= 80)
                     <*> (o A..:? "language_extensions" A..!= []);
           steps <- (o A..: "steps"
                     >>= fmap concat . mapM (parseSteps config));
           return (config {configSteps = steps}) }

src/Language/Haskell/Stylish/Config.hs:123:40:
    No instance for (Foldable t0) arising from a use of ‘concat’
    The type variable ‘t0’ is ambiguous
    Note: there are several potential instances:
      instance Foldable (Control.Applicative.Const m)
        -- Defined in ‘Control.Applicative’
      instance Foldable (Either a) -- Defined in ‘Data.Foldable’
      instance Foldable Data.Functor.Identity.Identity
        -- Defined in ‘Data.Functor.Identity’
      ...plus 96 others
    In the first argument of ‘fmap’, namely ‘concat’
    In the first argument of ‘(.)’, namely ‘fmap concat’
    In the second argument of ‘(>>=)’, namely
      ‘fmap concat . mapM (parseSteps config)’

src/Language/Haskell/Stylish/Config.hs:123:49:
    No instance for (Traversable t0) arising from a use of ‘mapM’
    The type variable ‘t0’ is ambiguous
    Note: there are several potential instances:
      instance Traversable (Control.Applicative.Const m)
        -- Defined in ‘Data.Traversable’
      instance Traversable (Either a) -- Defined in ‘Data.Traversable’
      instance Traversable Data.Functor.Identity.Identity
        -- Defined in ‘Data.Functor.Identity’
      ...plus 93 others
    In the second argument of ‘(.)’, namely ‘mapM (parseSteps config)’
    In the second argument of ‘(>>=)’, namely
      ‘fmap concat . mapM (parseSteps config)’
    In a stmt of a 'do' block:
      steps <- (o A..: "steps"
                >>= fmap concat . mapM (parseSteps config))
Failed to install stylish-haskell-0.5.11.0
```

**Comment by jaspervdj** [comment]
Could you try the latest master? I pushed a fix but don't have GHC 7.10 installed on this machine.

**Comment by NobbZ** [comment]
I will give it a try in the evening.
Am 16.01.2015 13:21 schrieb "Jasper Van der Jeugt" <notifications@github.com

**Comment by NobbZ** [comment]
In generall I am able to compile now, and a quickcheck with a very small file worked fine, but during build there are some warnings:

```
$ cabal build > /dev/null
src/Language/Haskell/Stylish/Config.hs:13:1: Warning:
    The import of ‘<*>, pure’
    from module ‘Control.Applicative’ is redundant

src/Language/Haskell/Stylish/Parse.hs:8:1: Warning:
    Module ‘Control.Monad.Error’ is deprecated:
      Use Control.Monad.Except instead

src/Language/Haskell/Stylish/Config.hs:13:1: Warning:
    The import of ‘<*>, pure’
    from module ‘Control.Applicative’ is redundant

src/Language/Haskell/Stylish/Parse.hs:8:1: Warning:
    Module ‘Control.Monad.Error’ is deprecated:
      Use Control.Monad.Except instead

src/Language/Haskell/Stylish/Config.hs:13:1: Warning:
    The import of ‘<*>, pure’
    from module ‘Control.Applicative’ is redundant

src/Language/Haskell/Stylish/Parse.hs:8:1: Warning:
    Module ‘Control.Monad.Error’ is deprecated:
      Use Control.Monad.Except instead
```

Ignore them or report in a new ticket?

**Comment by jaspervdj** [comment]
I have fixed these warnings and released a new version, `stylish-haskell-0.5.11.1`.
